framework/src/node/forger/forger.ts

```diff
@@ -191,15 +223,52 @@ export class Forger {
 			);
 		}
 
-		const account = await this._chainModule.dataAccess.getAccountByAddress(forgerAddress);
-
-		if (forging) {
-			this._keypairs.set(forgerAddress, keypair);
-			this._logger.info(`Forging enabled on account: ${account.address.toString('hex')}`);
-		} else {
+		if (!forging) {
+			// Disable delegate by removing keypairs corresponding to address
 			this._keypairs.delete(forgerAddress);
-			this._logger.info(`Forging disabled on account: ${account.address.toString('hex')}`);
+			this._logger.info(`Forging disabled on account: ${forgerAddress.toString('hex')}`);
+			return {
+				address: forgerAddress,
+				forging,
+			};
+		}
+
+		const previouslyForgedMap = await getPreviouslyForgedMap(this._db);
+		const forgingInput = { height, maxHeightPreviouslyForged, maxHeightPrevoted };
+		if (forgerInfoMissing(previouslyForgedMap, forgerAddress, forgingInput, overwrite)) {
+			throw new Error('Failed to enable forging due to missing forger info.');
+		}
+
+		if (isContradictingForgerInfo(previouslyForgedMap, forgerAddress, forgingInput, overwrite)) {
+			const expectedForgerInfo = previouslyForgedMap.get(forgerAddress) ?? {
+				height: 0,
+				maxHeightPreviouslyForged: 0,
+				maxHeightPrevoted: 0,
+			};
+			assert(
```

Comment:
If I understand correctly, instead of throwing the error explicitly I expect the assert to throw the error because of what I check in the if condition is the same as the first condition of the assert? If this is true, then I find it quite convoluted.

Reply:
mmm since throwing the error will the conflicting information was too much, so i added assert. but the assertion is only against height, maxHeightPreviouslyForged and maxHeightPrevoted. I can refactor to throw an error if its twisted

Reply:
I also dont understand the `assert` here, i think it's better to simply throw